### ISSUE TITLE
Training tips?

### ISSUE THREAD

**Issue by emaadmanzoor** [JieyuZ2/TMN#6]
Hi, thank you so much for releasing this code; it is really clean and works well!

I am trying to train TMN on my own taxonomy (I am the author of Arborist, which you compare with in your paper).

I was wondering if you had any tips for training to get the best performance? I am currently trying different values of:

   * k
   * Learning rate
   * Batch size
   * Number of negative samples per batch

Are there any other parameters you recommend I tune?

Thanks in advance!

**Comment by JieyuZ2** [comment]
Hey @emaadmanzoor 

Yeah I think you're tuning the right parameters and if I remember correctly, the activation function can also affect the performance. BTW in our more recent work, we had more complex model which may give better performance: https://arxiv.org/abs/2202.04887

Best regards,

**Comment by emaadmanzoor** [comment]
Awesome, thanks for sharing! I have been able to tune these hyperparameters with pretty good results, I will check out your improved model.
